src/Compilers/Core/Portable/RuntimeCapability.cs

```diff
@@ -0,0 +1,92 @@
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies capabilities that may or may not be supported by the common language runtime the compilation is
+    /// targeting.
+    /// </summary>
+    public enum RuntimeCapability
+    {
+        /// <summary>
+        /// Represents a runtime feature where types can define ref fields.
+        /// </summary>
+        ByRefFields = 1,
+
+        /// <summary>
+        /// Represents a runtime feature where overriding methods can return more derived types than the method they override.
+        /// </summary>
+        CovariantReturnsOfClasses = 2,
+
+        /// <summary>
+        /// Represents a runtime feature where interfaces can define default method implementations.
+        /// </summary>
+        DefaultImplementationsOfInterfaces = 3,
+
+        /// <summary>
+        /// Indicates that this version of the runtime supports IntPtr and UIntPtr as numeric types.
+        /// </summary>
+        NumericIntPtr = 4,
+
+        /// <summary>
+        /// Represents a runtime feature where C# function pointers can be declared with an unmanaged calling convention.
+        /// </summary>
+        UnmanagedSignatureCallingConvention = 5,
+
+        /// <summary>
+        /// Indicates that this version of runtime supports virtual static members of interfaces.
+        /// </summary>
+        VirtualStaticsInInterfaces = 6,
+    }
+
+    /// <summary>
+    /// Shared helpers to determine if a runtime capability is supported.  Located in a shared location so that we
+    /// return the same result per runtime, regardless of whether this is a C# or VB compilation.
+    /// </summary>
+    internal static class RuntimeCapabilityHelpers
+    {
+        public static bool RuntimeSupportsCapability(IAssemblySymbolInternal assembly, RuntimeCapability capability)
+        {
+            switch (capability)
+            {
+                case RuntimeCapability.ByRefFields:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__ByRefFields);
+
+                case RuntimeCapability.CovariantReturnsOfClasses:
+                    // check for the runtime feature indicator and the required attribute.
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__CovariantReturnsOfClasses) &&
+                           assembly.GetSpecialType(SpecialType.System_Runtime_CompilerServices_PreserveBaseOverridesAttribute) is { TypeKind: TypeKind.Class };
+
+                case RuntimeCapability.DefaultImplementationsOfInterfaces:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__DefaultImplementationsOfInterfaces);
+
+                case RuntimeCapability.NumericIntPtr:
+                    // CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol.
```

Comment:
>// CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol. [](http://example.com/codeflow?start=20&length=105)

I think this issue is specific to EnC scenarios only, caused by the fact how EnC creates symbols. Other consumers shouldn't be able to run into this problem. In addition, it looks very strange that we are protecting only this case.